### ISSUE TITLE
1.9.18 breaks {} as arguments default_value

### ISSUE THREAD

**Issue by swalkinshaw** [rmosolgo/graphql-ruby#2691]
The Ruby 2.7 compat changes in 1.9.18 have broken support for using a `default_value` of `{}` in arguments when building a schema from definition.

Backtrace:
```
[1] pry(main)> GraphQL::Schema.from_definition(schema_file)
ArgumentError: wrong number of arguments (given 1, expected 2)
from /Users/scottwalkinshaw/dev/graphql-ruby/lib/graphql/define/instance_definable.rb:307:in `call'
Exception: ArgumentError: wrong number of arguments (given 1, expected 2)
--
 0: /Users/scottwalkinshaw/dev/graphql-ruby/lib/graphql/define/instance_definable.rb:307:in `call'
 1: /Users/scottwalkinshaw/dev/graphql-ruby/lib/graphql/define/defined_object_proxy.rb:43:in `method_missing'
 2: /Users/scottwalkinshaw/dev/graphql-ruby/lib/graphql/define/instance_definable.rb:165:in `public_send'
 3: /Users/scottwalkinshaw/dev/graphql-ruby/lib/graphql/define/instance_definable.rb:165:in `block in ensure_defined'
 4: /Users/scottwalkinshaw/dev/graphql-ruby/lib/graphql/define/instance_definable.rb:156:in `each'
 5: /Users/scottwalkinshaw/dev/graphql-ruby/lib/graphql/define/instance_definable.rb:156:in `ensure_defined'
```

Reproduce:
```ruby
class MyInput < GraphQL::Schema::InputObject
  argument :foo, String, required: false
end

class Query < GraphQL::Schema::Object
  field :bar, String, null: false do
    argument :input, MyInput, required: false, default_value: {}
  end
  def bar(input:); 'foobar'; end
end

class DummySchema < GraphQL::Schema
  query(Query)
end

GraphQL::Schema.from_definition(DummySchema.to_definition)
```

Specifically it's this change that broke it: https://github.com/rmosolgo/graphql-ruby/compare/v1.9.17...v1.9.18

Here's the difference:

```ruby
irb(main):005:0> def with_kwargs(*args, **kwargs); puts args.inspect; puts kwargs.inspect; end
irb(main):007:0> with_kwargs
[]
{}
=> nil
irb(main):008:0> with_kwargs({})
[]
{}
=> nil
irb(main):009:0> def without_kwargs(*args); puts args.inspect; end
irb(main):010:0> without_kwargs({})
[{}]
```

The problem is `kwargs` is *always* an empty hash now and there's certain conditionals added for compat reason such as here: https://github.com/rmosolgo/graphql-ruby/blob/2f23a6d971ca1888387fb87848ad94962a955b49/lib/graphql/define/defined_object_proxy.rb#L38-L44

**Comment by rmosolgo** [comment]
🤦‍♂ Thanks. Hmmm...

**Comment by terencechow** [comment]
We are also getting the same error message and the version 1.9.18 causes a problem while 1.9.17 does not. However we do not use `default_value` and instead are using a [graphql function] (https://www.rubydoc.info/github/rmosolgo/graphql-ruby/GraphQL/Function). Hope this information helps!

**Comment by rmosolgo** [comment]
Thanks for sharing. As @swalkinshaw pointed out, our Ruby 2.7 "fixes" made it impossible to distinguish between "no keywords passed" and "empty object passed". It will be tricky to find a way to distinguish between those :S

**Comment by swalkinshaw** [comment]
Yeah my brain hurt after that debugging but I couldn't even think of a solution 😰

**Comment by terencechow** [comment]
> Thanks for sharing. As @swalkinshaw pointed out, our Ruby 2.7 "fixes" made it impossible to distinguish between "no keywords passed" and "empty object passed". It will be tricky to find a way to distinguish between those :S

I should share my schema as well. Note that I had a hard time understanding how my scenario was triggering this because we don't have any empty object or keywords passed...

Here's a sample of our schema:

```
class Child < GraphQL::Schema::Object
    field :id, Int, null: false
end

class Parent < GraphQL::Schema::Object
    field :id, Int, null: false

    field :children, Child.connection_type, null: true,
      function: FindRecords.new(type: Child.connection, property: :children)
end

class FindRecords < GraphQL::Function
  attr_reader :type

  def initialize(type:, property:)
    @property = property
    @type = type
  end

  def call(obj, args, ctx)
     obj.send(property)
  end
end
```

If you think of parent and children as active record models with a `has_many` relationship you can see what's going on here. (This is a snippet of our schema so I may have missed something while transcribing it)

**Comment by rmosolgo** [comment]
Thanks for reporting this issue. Turns out I could use the `ruby2_keywords` helper in Ruby 2.7 to make it work everywhere. I released 1.9.19 and 1.10.1 with the fix.

**Comment by swalkinshaw** [comment]
🎉 thanks! I didn't know about `ruby2_keywords`
